plugin/src/test/java/io/jenkins/plugins/redis/RedisConnectionTest.java

```diff
@@ -0,0 +1,177 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.plugins.redis;
+
+import eu.rekawek.toxiproxy.model.ToxicDirection;
+import hudson.Util;
+import hudson.model.Fingerprint;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.ToxiproxyContainer;
+import org.testcontainers.containers.Network;
+import redis.clients.jedis.exceptions.JedisException;
+
+import java.io.IOException;
+
+import static junit.framework.TestCase.fail;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNot.not;
+
+public class RedisConnectionTest {
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @Rule
+    public Network network = Network.newNetwork();
+
+    @Rule
+    public GenericContainer redis = new GenericContainer("redis:6.0.4-alpine")
+            .withExposedPorts(6379)
+            .withNetwork(network);
+
+    @Rule
+    public ToxiproxyContainer toxiproxy = new ToxiproxyContainer()
+            .withNetwork(network)
+            .withNetworkAliases("toxiproxy");
+
+    private void setRedisConfigurationViaProxy(ToxiproxyContainer.ContainerProxy proxy) {
+        final String host = proxy.getContainerIpAddress();
+        final int port = proxy.getProxyPort();
+        Utils.setRedisConfiguration(host, port);
+    }
+
+    @Test
+    public void testRedisConnectionFailureForSave() throws IOException {
+        String id = Util.getDigestOf("testRedisConnectionFailureForSave");
+
+        final ToxiproxyContainer.ContainerProxy proxy = toxiproxy.getProxy(redis, 6379);
+        setRedisConfigurationViaProxy(proxy);
+        proxy.setConnectionCut(true);
+
+        try {
+            new Fingerprint(null, "foo.jar", Util.fromHexString(id));
+        } catch (JedisException e) {
+            proxy.setConnectionCut(false);
+            Fingerprint fingerprintLoaded = Fingerprint.load(id);
+            assertThat(fingerprintLoaded, is(nullValue()));
```

Comment:
Do we need need to assert the error message?

Reply:
So i did not do something like `@Test(expected=JedisException.class)` because I want to go beyond the error and check if everything works when connection is reestablished

Reply:
That is a good one, but what error message does the user see from JedisException. Generally, you can assert that the exception class was thrown *and* the message. See this example for junit4: 
https://www.baeldung.com/junit-assert-exception#junit-4 

You can probably follow the example pattern to make it one test and make it easier to read